pydata_center/monitoring/views.py

```diff
@@ -270,3 +267,98 @@ def dashboard_view(request):
         template_name='monitoring/dashboard.html',
         context={'agents': agents}
     )
+
+
+@extend_schema(
+    tags=['Metrics'],
+    parameters=[
+        OpenApiParameter(
+            name='hostname',
+            type=OpenApiTypes.STR,
+            location=OpenApiParameter.QUERY,
+            description='Hostname of the server to filter metrics.'
+        ),
+        OpenApiParameter(
+            name='start',
+            type=OpenApiTypes.DATETIME,
+            location=OpenApiParameter.QUERY,
+            description='Start datetime (ISO 8601) for metrics filtering.'
+        ),
+        OpenApiParameter(
+            name='end',
+            type=OpenApiTypes.DATETIME,
+            location=OpenApiParameter.QUERY,
+            description='End datetime (ISO 8601) for metrics filtering.'
+        ),
+    ],
+    responses={
+        status.HTTP_200_OK: OpenApiResponse(
+            description='List of filtered agent metrics.'
+        ),
+        status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+            description='Invalid datetime format or query parameters.'
+        ),
+    },
+    description=(
+        'Returns historical server metrics (CPU, RAM, disk usage, '
+        'load average) based on optional filters: hostname, '
+        'start time, and end time.'
+    ),
+)
+@api_view(['GET'])
+def metrics_history_view(request):
+    """
+    Returns historical server metrics as JSON.
+    Processes a GET request with optional `hostname`, `start`, and `end`
+    parameters. Filters the ServerStatus records based on the provided
+    criteria and returns a list of metrics (CPU, RAM, disk usage, load
+    average) within the specified time range.
+    """
+    hostname = request.GET.get('hostname')
+    start_str = request.GET.get('start')
+    end_str = request.GET.get('end')
+
+    start = parse_datetime(start_str) if start_str else None
+    end = parse_datetime(end_str) if end_str else None
```

Comment:
Currently, `parse_datetime()` silently returns `None` if the input is invalid. This means incorrect formats (like "abc") will not raise errors, and you'll return all metrics instead, silently ignoring user error.
Add validation logic:
```
from rest_framework.exceptions import ValidationError

if start_str and not start:
    raise ValidationError({'start': 'Invalid datetime format.'})
if end_str and not end:
    raise ValidationError({'end': 'Invalid datetime format.'})
```